### PR TITLE
Remove upper version constraint on hashicorp/random

### LINKED ISSUE #51: hashicorp/random version constraint < 3.0.0 is a pain for Apple silicon users
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* ***If you are interested in working on this issue or have submitted a pull request, please leave a comment!***

***

### Summary

We're using terraform-aws-iam-role@4.0.0. It has a constraint on `hashicorp/random` of `>= 1.2.0, < 3.0.0`. I'm over here on Apple  Silicon and it's not until `3.x` that a `darwin_arm64` provider was made available. 

https://github.com/traveloka/terraform-aws-iam-role/blob/master/modules/crossaccount/versions.tf
https://github.com/traveloka/terraform-aws-iam-role/blob/master/modules/external/versions.tf
https://github.com/traveloka/terraform-aws-iam-role/blob/master/modules/instance/versions.tf
https://github.com/traveloka/terraform-aws-iam-role/blob/master/modules/lambda/versions.tf

Currently, we've forked and removed the upper constraints. No problems for us, using `hashicorp/random v3.1.2`

What do you think? I'm not sure about the original reason for the upper constraint.

```
hashicorp/aws v3.75.1
```

### PR DESCRIPTION
**Pull request by slinkdarga**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

***

Fixes #51

### CODE REVIEW

**Comment by salvianreynaldi** [comment]
+1 this,
probably there's still a value in retaining the upper limit, e.g. to version `3.x`, as 4.x might break things

**Comment by salvianreynaldi** [comment]
v3.0.0's breaking change is only [the removal of the `b64` attribute](https://github.com/hashicorp/terraform-provider-random/commit/486d27e8730b6f3a6dc9cec27aebbea8596f3422), which is not used in this repo, so it should be safe to use.

@slinkdarga could you update the PR to add 3.x as the (inclusive) upper limit as above?

File: modules/crossaccount/versions.tf [hunk 1]
```diff
@@ -5,7 +5,7 @@ terraform {
     }
     random = {
       source  = "hashicorp/random"
-      version = ">= 1.2.0, < 3.0.0"
```
Comment (vincenttjia) [inline modules/crossaccount/versions.tf:8]: Please add an upper version limit

```
version = ">= 1.2.0, < 4.0.0"
```

File: modules/external/versions.tf [hunk 2]
```diff
@@ -5,7 +5,7 @@ terraform {
     }
     null = {
       source  = "hashicorp/null"
-      version = ">= 1.0.0, < 3.0.0"
```
Comment (vincenttjia) [inline modules/external/versions.tf:8]: Please add an upper version limit

```
version = ">= 1.2.0, < 4.0.0"
```

File: modules/instance/versions.tf [hunk 3]
```diff
@@ -7,7 +7,7 @@ terraform {
     }
     random = {
       source  = "hashicorp/random"
-      version = ">= 1.2.0, < 3.0.0"
```
Comment (vincenttjia) [inline modules/instance/versions.tf:10]: Please add an upper version limit

```
version = ">= 1.2.0, < 4.0.0"
```

File: modules/lambda/versions.tf [hunk 4]
```diff
@@ -5,7 +5,7 @@ terraform {
     }
     random = {
       source  = "hashicorp/random"
-      version = ">= 1.2.0, < 3.0.0"
```
Comment (vincenttjia) [inline modules/lambda/versions.tf:8]: Please add an upper version limit

```
version = ">= 1.2.0, < 4.0.0"
```

File: modules/service/versions.tf [hunk 5]
```diff
@@ -7,7 +7,7 @@ terraform {
     }
     random = {
       source  = "hashicorp/random"
-      version = ">= 1.2.0, < 3.0.0"
```
Comment (vincenttjia) [inline modules/service/versions.tf:10]: Please add an upper version limit

```
version = ">= 1.2.0, < 4.0.0"
```

**Review by vincenttjia — CHANGES_REQUESTED** [review]

**Comment by salvianreynaldi** [comment]
Hi @vincenttjia,
I've updated the line, kindly review 🙇

**Review by vincenttjia — APPROVED** [review]
LGTM

**Review by vincenttjia — DISMISSED** [review]

**Pull request merged**